crates/fluvio-protocol/benches/bench.rs

```diff
@@ -0,0 +1,71 @@
+#![feature(test)]
+
+extern crate test;
+
+#[cfg(test)]
+mod bench {
+    use std::io::Cursor;
+
+    use fluvio_protocol::{Decoder, Encoder};
+    use test::{Bencher, black_box};
+
+    #[bench]
+    fn bench_decode_bool(b: &mut Bencher) {
+        b.iter(|| {
+            for i in 1..100 {
+                let data = [0x1];
+                let mut value: bool = false;
+
+                black_box(|| {
+                    value.decode(&mut Cursor::new(&data), 0);
+                });
+            }
+        });
+    }
+
+    #[bench]
+    fn bench_encode_bool(b: &mut Bencher) {
+        b.iter(|| {
+            for i in 1..100 {
+                let mut dest = vec![];
+                let value = true;
+
+                black_box(|| {
+                    value.encode(&mut dest, 0);
+                });
+            }
+        });
+    }
+
+    #[bench]
+    fn bench_encode_vecu8(b: &mut Bencher) {
```

Comment:
Could you instead or optionally benchmark instead SmartModuleSpec?

Reply:
@morenol actually, what do you think about introducing more benchmarks on a separate PR? One where we could focus on adding benchmarks? That way I can speed up progress on #2738?

Reply:
Yes, can always add more benchmarks later.